File: Kotlin/프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.kt
import java.util.*;
class Solution {
    fun solution(k: Int, score: IntArray): IntArray {
        var answer: IntArray = IntArray(score.size)
        val pq = PriorityQueue<Int> { a, b -> a - b }
        for ((i, e) in score.withIndex()) {
            pq.add(e)
            if (pq.size > k) {
                pq.poll()
            }
            answer[i] = pq.peek()
        }
        return answer
    }
}

File: Python3/프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq
def solution(k, score):
    answer = []
    heap = []
    for e in score:
        heapq.heappush(heap, e)
        if len(heap) > k:
            heapq.heappop(heap)
        answer.append(heap[0])

    return answer